### PR TITLE
AB#25769 Bug on validations in currency custom field

### PR DESCRIPTION
**Pull request by aurelio-aot**

### CODE REVIEW

File: applications/Unity.GrantManager/src/Unity.GrantManager.Web/Views/Shared/Components/AssessmentResults/Default.js [hunk 1]
```diff
@@ -190,13 +190,35 @@ function flaggedFieldIsValid(flag, name) {
...
+                invalidFieldsFound = true;
+            }
         }
+        
+    });
+
+    return invalidFieldsFound;
+}
+
+function formHasInvalidCurrencyCustomFields(formId) {
```
Comment (AndreGAot) [inline applications/Unity.GrantManager/src/Unity.GrantManager.Web/Views/Shared/Components/AssessmentResults/Default.js:211]: these methods should be moved to the GrantApplications/details.js file as intent is to share for all the child tabs not just the AssessmentResults tab (what if we don't load the assessment results tab?)

File: applications/Unity.GrantManager/src/Unity.GrantManager.Web/Pages/GrantApplications/Details.js [hunk 2]
```diff
@@ -293,9 +293,16 @@ $(function () {
...
+        setTimeout(function () {
+            $('.custom-currency-input').maskMoney();
+        }, 1000);
+    }
+    
     PubSub.subscribe('application_assessment_results_saved',
         (msg, data) => {
-            assessmentResultWidgetManager.refresh();        
+            assessmentResultWidgetManager.refresh();
+            initCustomFieldCurrencies();
```
Comment (AndreGAot) [inline applications/Unity.GrantManager/src/Unity.GrantManager.Web/Pages/GrantApplications/Details.js:305]: there are other places where we refresh the assessmentResultWidgetManager.refresh but don't subsequently call the initCustomFieldCurrencies

File: applications/Unity.GrantManager/src/Unity.GrantManager.Web/Pages/GrantApplications/Details.js [hunk 3]
```diff
@@ -293,9 +293,16 @@ $(function () {
             assessmentResultWidgetManager.refresh();            
         }
     );
+    function initCustomFieldCurrencies() {
+        setTimeout(function () {
+            $('.custom-currency-input').maskMoney();
+        }, 1000);
```
Comment (AndreGAot) [inline applications/Unity.GrantManager/src/Unity.GrantManager.Web/Pages/GrantApplications/Details.js:299]: this is always ominous and sounds like a race condition, what happens if we make this time 100ms instead?

**Review by AndreGAot — CHANGES_REQUESTED** [review]
There is some more refactoring and research needed to get this right - I also ended up along the data-min / data-max path when trying to solve this - Ideally we want some other ways to trigger the rebind of the moneymask which is where I got stuck too

**Comment by aurelio-aot** [comment]
hi @AndreGAot -- this is now ready for another round of review. thanks.

File: applications/Unity.GrantManager/src/Unity.GrantManager.Web/Views/Shared/Components/AssessmentResults/Default.js (hunk 1 above)
Comment (aurelio-aot) [reply applications/Unity.GrantManager/src/Unity.GrantManager.Web/Views/Shared/Components/AssessmentResults/Default.js]: @AndreGAot - done moving the function to GrantApplications/Details.js

File: applications/Unity.GrantManager/src/Unity.GrantManager.Web/Pages/GrantApplications/Details.js (hunk 2 above)
Comment (aurelio-aot) [reply applications/Unity.GrantManager/src/Unity.GrantManager.Web/Pages/GrantApplications/Details.js]: @AndreGAot - modified this code to use MutationObserver

File: applications/Unity.GrantManager/src/Unity.GrantManager.Web/Pages/GrantApplications/Details.js (hunk 3 above)
Comment (aurelio-aot) [reply applications/Unity.GrantManager/src/Unity.GrantManager.Web/Pages/GrantApplications/Details.js]: @AndreGAot - modified this code to use MutationObserver.

**Comment by aurelio-aot** [comment]
hi @AndreGAot - i pushed code to display comma separator of currencies on initial load, this is ready again for another round of review. thanks.

**Review by jimmyPasta — APPROVED** [review]

**Review by AndreGAot — APPROVED** [review]
Ready for Dev

**Pull request merged**